README.md

```diff
@@ -14,31 +15,38 @@ Via Composer
 $ composer require slim/csrf
 ```
 
-Requires Slim 3.0.0 or newer.
+Requires Slim 4.0.0 or newer.
 
 ## Usage
 
-In most cases you want to register Slim\Csrf for all routes, however,
-as it is middleware, you can also register it for a subset of routes.
-
+In most cases you want to register Slim\Csrf for all routes, however, as it is middleware, you can also register it for a subset of routes.
 
 ### Register for all routes
 
 ```php
+use DI\Container;
+use Slim\Csrf\Guard;
+use Slim\Factory\AppFactory;
+
+require __DIR__ . '/vendor/autoload.php';
+
 // Start PHP session
 session_start();
 
-$app = new \Slim\App();
+// Create Container
+$container = new Container();
 
-// Register with container
-$container = $app->getContainer();
-$container['csrf'] = function ($c) {
-    return new \Slim\Csrf\Guard;
-};
+// Create App
+$app = AppFactory::create();
+$responseFactory = $app->getResponseFactory();
+
+// Register Middleware On Container
+$container->set('csrf', function () use ($responseFactory) {
+    return new Guard($responseFactory);
+});
 
-// Register middleware for all routes
-// If you are implementing per-route checks you must not add this
-$app->add($container->get('csrf'));
+// Register Middleware To Be Executed On All Routes
+$app->add('csrf');
 
 $app->get('/foo', function ($request, $response, $args) {
     // CSRF token name and value
```

Comment:
The next two lines access `$this->csrf` directly. I guess that this is no more possible in Slim 4.
```php
    /** @var Guard $csrf */
    $csrf = $this->get('csrf');

    $nameKey = $csrf->getTokenNameKey();
    $valueKey = $csrf->getTokenValueKey();
```

Reply:
This is still possible, we re-added container binding to callables as of 4.2

Reply:
That is weird. If I am using php-di 6.0.9 and Slim 4.x-dev 9c18e47 (current dev version) I get the following error message
```
Notice: Undefined property: DI\Container::$csrf in ...\public\index.php on line 28
```
and on line 28 I have
```php
    $nameKey = $this->csrf->getTokenNameKey();
```

A quick `die(get_class($this));` right before line 28 gives me `DI\Container`. So yes, the binding did work, but php-di does not support the magic `__get`. Or am I missing anything else?